test/util/IdTableHelpers.h

```diff
@@ -75,47 +78,73 @@ inline IdTable makeIdTableFromVector(const VectorTable& content,
+generated row must ALWAYS have size `numberColumns`. Otherwise an exception will
+be thrown.
+*/
+IdTable generateIdTable(const size_t numberRows, const size_t numberColumns,
+                        const std::function<IdTable::row_type()>& rowGenerator);
 
-  if (resultMustBeSortedByJoinColumn) {
-    // Is the table sorted by join column?
-    ASSERT_TRUE(std::ranges::is_sorted(localTable.getColumn(joinColumn)));
-  }
+/*
+@brief Creates a `IdTable`, where the content of the join column is given via
+functions and all other columns are randomly filled with numbers.
 
-  // Sort both the table and the expectedContent, so that both have a definite
-  // form for comparison.
-  std::ranges::sort(localTable, std::ranges::lexicographical_compare);
-  std::ranges::sort(localExpectedContent, std::ranges::lexicographical_compare);
+@param numberRows numberColumns The number of rows and columns, the table should
+have.
+@param joinColumnWithGenerator Every pair describes the position of a join
+column and the function, which will be called, to generate it's entries.
+*/
+IdTable createRandomlyFilledIdTable(
+    const size_t numberRows, const size_t numberColumns,
+    const std::vector<std::pair<size_t, std::function<ValueId()>>>&
+        joinColumnWithGenerator);
 
-  ASSERT_EQ(localTable, localExpectedContent);
-}
+/*
+@brief Creates a `IdTable`, where the content of the join columns is given via
+a function and all other columns are randomly filled with numbers.
+
+@param numberRows numberColumns The number of rows and columns, the table should
+have.
+@param joinColumns The join columns.
+@param generator The generator for the join columns. Order of calls: Row per
+row, starting from row 0, and in a row for every join column, with the join
+columns ordered by their column. Starting from column 0.
+*/
+IdTable createRandomlyFilledIdTable(const size_t numberRows,
+                                    const size_t numberColumns,
+                                    const std::vector<size_t>& joinColumns,
+                                    const std::function<ValueId()>& generator);
+
+/*
+ * @brief Return a IdTable, that is randomly filled. The range of numbers
+ *  being entered in the join column can be defined.
+ *
+ * @param numberRows, numberColumns The size of the IdTable, that is to be
+ *  returned.
+ * @param joinColumn The joinColumn of the IdTable, that is to be returned.
+ * @param joinColumnLowerBound, joinColumnUpperBound The range of the entries
+ *  in the join column, definied as
+ *  [joinColumnLowerBound, joinColumnUpperBound].
+ */
+IdTable createRandomlyFilledIdTable(const size_t numberRows,
+                                    const size_t numberColumns,
+                                    const size_t joinColumn,
+                                    const size_t joinColumnLowerBound,
+                                    const size_t joinColumnUpperBound);
```

Comment:
I would like to have this function also as a variant for multiple join columns with multiple generators, so you
have a struct `JoinColumnAndBounds`, and then this function has arguments
`numRows, numColumns, std::vector<JoinColumnAndBounds>`

Reply:
Done, see [e797d79](https://github.com/ad-freiburg/qlever/pull/1016/commits/e797d79d476ed956a481c248f6749e5148c2d124).